packages/cli-kit/src/session/exchange.ts

```diff
@@ -127,7 +157,7 @@ async function requestAppToken(
 }
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-async function tokenRequest(params: {[key: string]: string}): Promise<any> {
+async function tokenRequest(params: {[key: string]: string}, throws = true): Promise<any> {
```

Comment:
Having two throwing behaviours within the same function controlled via function argument is a bit uncommon. Also, the API is not very idiomatic: 

```js
await tokenRequest(params, false)
```

I'd consider removing that argument and use the [new Result type](https://github.com/Shopify/cli/pull/333) to let the caller decide what to do with the error.